xbmc/platform/posix/filesystem/SMBFile.cpp

```diff
@@ -29,14 +29,35 @@
 
 #include <inttypes.h>
 #include <mutex>
+#include <regex>
 
 #include <libsmbclient.h>
 
 using namespace XFILE;
 
-void xb_smbc_log(const char* msg)
+void xb_smbc_log(void* private_ptr, int level, const char* msg)
 {
-  CLog::Log(LOGINFO, "{}{}", "smb: ", msg);
+  const int logLevel = [level]()
+  {
+    switch (level)
+    {
+      case 0:
+        return LOGWARNING;
+      case 1:
+        return LOGINFO;
+      default:
+        return LOGDEBUG;
+    }
+  }();
+
+  if (strchr(msg, '@'))
+  {
+    // redact User/pass in URLs
+    std::regex redact("(\\w+://)\\S+:\\S+@");
```

Comment:
```suggestion
    static const std::regex redact("(\\w+://)\\S+:\\S+@");
```
This should take care of a part of your concern:

> reduce the load on RE compiler/executer

Keeping the `strchr` in addition doesn't hurt :slightly_smiling_face:

Reply:
Nice touch. Thank you.